app/controllers/automated_tests_controller.rb

```diff
@@ -30,6 +30,8 @@ def update
   def manage
     @assignment = Assignment.find(params[:assignment_id])
     @assignment.test_groups.build
+    find_or_create_test_student(params[:assignment_id])
```

Comment:
why not have the `find_or_create_test_student` method return the grouping and then you don't need the query on the next line. If you do you might want to rename it to `find_or_create_test_grouping` so that its more obvious what the return value is.